<a href="https://colab.research.google.com/github/rafradio/pandas_111122/blob/main/Train.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd

In [3]:
df = pd.read_csv('/content/train (1).csv')

In [4]:
df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [5]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

In [6]:
df.query("Survived == 1")

PassengerId  Survived  Pclass  \
1              2         1       1   
2              3         1       3   
3              4         1       1   
8              9         1       3   
9             10         1       2   
..           ...       ...     ...   
875          876         1       3   
879          880         1       1   
880          881         1       2   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8    Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                  Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
..                                                 ...     ...   ...    ...   
875                   Najib, Miss. Adele Kiamie "Jane"  female  15.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
880       Shelley, Mrs. William (Imanita Parrish Hall)  female  25.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch            Ticket     Fare Cabin Embarked  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
8        2            347742  11.1333   NaN        S  
9        0            237736  30.0708   NaN        C  
..     ...               ...      ...   ...      ...  
875      0              2667   7.2250   NaN        C  
879      1             11767  83.1583   C50        C  
880      1            230433  26.0000   NaN        S  
887      0            112053  30.0000   B42        S  
889      0            111369  30.0000  C148        C  

[342 rows x 12 columns]

In [ ]:
df.shape

(891, 12)

In [7]:
df.query("Survived == 1").shape[0]/df.shape[0]

0.3838383838383838

Выжило 38.38% от всех пассажиров

In [ ]:
df.query("Sex == 'female'").agg({'Survived': 'sum'})

Survived    233
dtype: int64

In [9]:
df.query("Sex == 'female'").agg({'Survived': 'sum'})/df.query("Sex == 'female'").shape[0]

Survived    0.742038
dtype: float64

Среди женщин выжило 74%

In [10]:
df.query("Sex == 'male'").agg({'Survived': 'sum'})/df.query("Sex == 'male'").shape[0]

Survived    0.188908
dtype: float64

Среди мужчин выжило только 18.8%

In [ ]:
df.query("Pclass == 3 and Sex == 'female'").shape

(144, 12)

In [ ]:
df.agg({'Fare': 'sum'})

Fare    28693.9493
dtype: float64

Общая выручка от продажи билетов - 28698.94 фунта стерлингов

In [12]:
df.query("Pclass == 3").agg({'Fare': 'sum'})/df.agg({'Fare': 'sum'})

Fare    0.234011
dtype: float64

Третий класс принес только 23.4% выручки. в то время масс-продажи не были развиты

In [13]:
df.query("Pclass == 1").agg({'Fare': 'sum'})/df.agg({'Fare': 'sum'})

Fare    0.633493
dtype: float64

Первый класс принес 63.3%. в то время маркетинг был рассчитан на богатых

In [ ]:
z = df.query("Pclass == 1 and Sex == 'male' and Survived == 1").agg({'Age': 'sum'}) / df.query("Pclass == 1 and Sex == 'male' and Survived == 1").shape[0]

In [ ]:
print(z)

Age    32.220444
dtype: float64


Средний возраст мужчин выэивших в первом классе

In [11]:
df.query("Pclass == 1 and Sex == 'male' and Survived == 1")['Age'].mean()

36.248000000000005

Средний возраст мужчин выживших в первом классе через метод mean()

In [ ]:
df.query("Pclass == 3 and Sex == 'female'")['Age'].mean()

21.75

Средний возраст женщин в третьем классе -21.75 лет

In [15]:
df.query("Pclass == 3 and Sex == 'male'")['Age'].mean()

26.507588932806325

Средний возраст мужчин в третьем классе - 26.5 лет. В третьем классе в основном ехали молодые люди, в поисках работы в новом свете.

In [22]:
df.query("Pclass == 3 and Age < 13 and Survived == 0").shape[0]/df.query("Pclass == 3 and Age < 13").shape[0]

0.5833333333333334

погибло 58% детей в третьем классе

In [24]:
df.query("Pclass == 3 and Age < 13").shape[0]/df.query("Pclass == 3").shape[0]

0.09775967413441955

В третьем классе было 9.7% детей

In [27]:
df.query("Pclass == 1 and Age < 13").shape[0]/df.query("Pclass == 1").shape[0]

0.018518518518518517

детей в первом классе было 1.8%

In [29]:
df.query("Pclass == 1 and Age < 13 and Survived == 1").shape[0]/df.query("Pclass == 1 and Age < 13").shape[0]

0.75

в первом классе погибло 25% детей

In [14]:
df.query("Sex == 'female' and Pclass == 1 and Survived == 0")

PassengerId  Survived  Pclass  \
177          178         0       1   
297          298         0       1   
498          499         0       1   

                                                Name     Sex   Age  SibSp  \
177                       Isham, Miss. Ann Elizabeth  female  50.0      0   
297                     Allison, Miss. Helen Loraine  female   2.0      1   
498  Allison, Mrs. Hudson J C (Bessie Waldo Daniels)  female  25.0      1   

     Parch    Ticket      Fare    Cabin Embarked  
177      0  PC 17595   28.7125      C49        C  
297      2    113781  151.5500  C22 C26        S  
498      2    113781  151.5500  C22 C26        S

В первом классе из 94 женщин погибло только 3 человека. Среди них мама с 2-х летним ребенком. 